pdc/apps/common/viewsets.py

```diff
@@ -63,6 +65,52 @@ def perform_create(self, serializer):
                                        _dumps_json(item.export()))
 
 
+class NotificationMixin(object):
+    def get_object(self):
+        obj = super(NotificationMixin, self).get_object()
+        serializer_class = self.get_serializer_class()
+        self.initial_status = serializer_class(instance=obj,
+                                               context={'request': self.request}).data
+        return obj
+
+    def _get_url(self, basename):
+        if not hasattr(self, 'object') or not self.object:
+            return None
+        # TODO how reliable is this?
+        return reverse('%s-detail' % basename, args=[getattr(self.object, self.lookup_field)])
```

Comment:
Probably unreliable, there can be more lookup fields.

Can you use [self.request.build_absolute_uri()](https://docs.djangoproject.com/en/1.8/ref/request-response/#django.http.HttpRequest.build_absolute_uri)?

Reply:
That works for update, but not for create we would get the url for the whole collection, not for the new object.

Reply:
This is basically what I try to use for auto-generating API documentation:

```python
def lookup_field_names(view):
    if hasattr(view, 'lookup_fields'):
        return [field for field, _ in view.lookup_fields]

    if hasattr(view, 'lookup_field'):
        return [view.lookup_field]

    return []
```

Reply:
This is pretty much what I ended up doing in the last commit. Having simple numeric IDs sure looks much more pleasant…